doc/languages-frameworks/php.section.md

```diff
@@ -153,3 +155,148 @@ php.override {
   };
 }
 ```
+
+### Building PHP projects {#ssec-building-php-projects}
```

Comment:
Please take this with caution, as I'm still pretty new to Nix. The following is nothing new, as we probably have the same situation with `phar` files already. But the new approach opens the possibility to use `composer audit`:

Would it make sense to have the testPhase executing `composer audit`? I'm not sure whether all people are aware of the effect of providing a `composer.lock` for software. It would result in locked dependencies, shifting the security updates from users to the maintainers. Not everyone is used to Nix, some might only know `composer req --global` where the user is in charge and could do security updates.

I expect a growing list of PHP derivations thanks to making it so easy to package PHP projects with the new approach. I fear some might not be aware of the security maintenance shift when maintainer bundle the software (providing phar or composer.lock). Leading to insecure software distribution via nix. A `composer audit` might help here to prevent building unsecure versions. Maintainers of the nix packages could provide PRs upstream to update `composer.lock` files in order to keep Nix packages secure.

What do you think? This could be opt-out, e.g. by `runComposerAudit = true` being the default, allowing users to remove the audit step for individual packages.

Could also be an improvement or another issue as follow-up to the current PR, in order to not block your awesome work. I'm really looking forward to having the new builder in nixpkgs.

Reply:
Thanks Daniel! I will add it to the checkphase and provide an option to make it optional.

Reply:
I think that requires network access so it will not really work in `checkPhase`. And since it would change over time, doing it within a derivation is not really a good place from reproducibility point of view. Probably best left for some external tool like vulnix.

And yeah, this is a downside of per-project dependencies compared to having a single distro-wide version of a package. In the worst case, we can apply patches to the lockfile, which was not possible with PHARs.

Reply:
Thanks Jan. We can even override composer.json now if needed.